pkg/kubeletclient/kubeletclient.go

```diff
@@ -38,10 +39,14 @@ func LocalEndpoint(path, file string) (string, error) {
 	return filepath.Join(u.String(), file+".sock"), nil
 }
 
+func removeUnixProtocol(endpoint string) (string) {
+	return strings.TrimPrefix(endpoint, (unixProtocol + ":"))
+}
+
```

Comment:
How about to use `u, err := url.Parse(endpoint)` as `hasKubeletAPIEndpoint()` does?

```
func removeUnixProtocol(endpoint string) (string, error) {
    u, err := url.Parse(endpoint)
    if err != nil {
        return "", err
    }
   return u.Path, nil
}
```